x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -80,17 +80,21 @@ export class EnterpriseSearchRequestHandler {
 
         // Call the Enterprise Search API
         const apiResponse = await fetch(url, { method, headers, body });
+        const { status } = apiResponse;
 
         // Handle response headers
         this.setResponseHeaders(apiResponse);
 
-        // Handle authentication redirects
-        if (apiResponse.url.endsWith('/login') || apiResponse.url.endsWith('/ent/select')) {
+        // Handle unauthenticated users / authentication redirects
+        if (
+          apiResponse.status === 401 ||
```

Comment:
Since this is destructured above, we can get rid of `apiResponse.` here. I'd also submit that you can also destructure `url` and clan this up a bit below